flags/flags.go

```diff
@@ -10,16 +10,93 @@
 //
 // SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
 
-//go:build !things
-// +build !things
-
 package flags
 
 import (
 	"flag"
 
+	"github.com/eclipse-kanto/suite-connector/flags"
+	"github.com/eclipse-kanto/suite-connector/logger"
+
 	"github.com/eclipse-kanto/azure-connector/config"
 )
 
-// AddHub adds the Hub connection related flags.
-func AddHub(_ *flag.FlagSet, _, _ *config.AzureSettings) {}
+const (
+	flagCACert           = "cacert"
+	flagTenantID         = "tenantId"
+	flagIDScope          = "idScope"
+	flagSASTokenValidity = "sasTokenValidity"
+)
+
+// AddGlobal adds the Cloud Agent global flags.
+func AddGlobal(f *flag.FlagSet) (configFile *string) {
+	return flags.AddGlobal(f)
+}
+
+// ConfigCheck checks for config file existence.
+func ConfigCheck(logger logger.Logger, configFile string) {
+	flags.ConfigCheck(logger, configFile)
+}
+
+// Add adds the Cloud Agent flags and uses the provided settings to collect the provided values.
+func Add(f *flag.FlagSet, settings *config.AzureSettings) {
+	def := config.DefaultSettings()
+
+	f.StringVar(&settings.TenantID,
+		flagTenantID, def.TenantID,
+		"Tenant ID",
+	)
+	f.StringVar(&settings.ConnectionString,
+		"connectionString", def.ConnectionString,
+		"The connection string for connectivity to Azure IoT Hub",
+	)
+	f.StringVar(&settings.SASTokenValidity,
+		flagSASTokenValidity, def.SASTokenValidity,
+		"The validity period for the generated SAS token for device authentication",
+	)
+	f.StringVar(&settings.PassthroughTelemetryTopic,
+		"passthroughTelemetryTopic", def.PassthroughTelemetryTopic,
+		"The list of passthrough telemetry topics the cloud agent listens to.",
+	)
+	f.StringVar(&settings.PassthroughCommandTopic,
+		"passthroughCommandTopic", def.PassthroughCommandTopic,
+		"The passthrough command topic where all messages from the cloud are forwarded to.",
+	)
+	f.StringVar(&settings.IDScope, flagIDScope, def.IDScope,
+		"ID Scope from Azure Device Provisioning Service",
```

Comment:
- ID Scope -> ID scope
- Azure Device Provisioning Service -> Azure Device Provisioning service
- `from` - I guess that it's `for`